frontend/pages/configurator.vue

```diff
@@ -0,0 +1,1160 @@
+    circle_size: circleSize.value,
+    circle_color: circleColor.value,
+    orientation: modalOrientation.value,
+    time_visible_ms: 5000,
+  };
+
+  const token = localStorage.getItem("token");
+
+  if (isEditing.value && editingIndex.value !== null) {
+    // Bearbeiten eines bestehenden Items
+    const item = testItems.value[editingIndex.value];
+    const oldItemId = item.id;
+
+    try {
+      let response;
+      if (!item.id || item.createdByUser) {
+        // POST-Request für ein neues Item
+        response = await $fetch("http://localhost:8000/api/item_configs", {
+          method: "POST",
+          body: updatedItem,
+          headers: { Authorization: `Bearer ${token}` },
+        });
+
+        // Aktualisieren der neuen ID und `selectedItems`
+        item.id = response.id;
+        item.isUnsaved = false;
+        const index = selectedItems.value.indexOf(oldItemId);
+        if (index !== -1) {
+          selectedItems.value.splice(index, 1, response.id);
+        }
+      } else {
+        // PUT-Request für ein bereits gespeichertes Item
+        await $fetch(`http://localhost:8000/api/item_configs/${item.id}`, {
+          method: "PUT",
+          body: updatedItem,
+          headers: { Authorization: `Bearer ${token}` },
+        });
+      }
+
+      // Aktualisiere das bearbeitete Item in `testItems`
+      testItems.value[editingIndex.value] = { ...updatedItem, id: item.id };
+      editingIndex.value = null;
+      isItemModalOpen.value = false;
+    } catch (error) {
+      console.error("Fehler beim Speichern/Aktualisieren:", error);
+    }
+  } else {
+    // Neues Item erstellen und in `testItems` und `selectedItems` hinzufügen
+    try {
+      const response = await $fetch("http://localhost:8000/api/item_configs", {
+        method: "POST",
+        body: updatedItem,
+        headers: { Authorization: `Bearer ${token}` },
+      });
+      testItems.value.push({ ...updatedItem, id: response.id });
+      selectedItems.value.push(response.id);
+      isItemModalOpen.value = false;
+    } catch (error) {
+      console.error("Fehler beim Erstellen eines neuen Items:", error);
+    }
+  }
+};
+
+const loadedTestId = ref(null); // Referenz für die aktuell geladene Test-ID
+
+// Neue Methode für die direkte Aktualisierung der Konfiguration
+const updateTestConfig = async () => {
+  try {
+    const token = localStorage.getItem("token");
+    const updateUrl = `${saveTestEndpoint}/${loadedTestId.value}`;
+
+    await $fetch(updateUrl, {
+      method: "PUT",
+      body: {
+        name: loadedTestName.value, // Der bearbeitbare Name im Input-Feld
+        item_config_ids: selectedItems.value,
+      },
+      headers: {
+        Authorization: `Bearer ${token}`,
+      },
+    });
+
+    // Zeige den Erfolg-Alert an
+    showSuccessAlert.value = true;
+
+    // Verstecke den Alert nach 3 Sekunden
+    setTimeout(() => {
+      showSuccessAlert.value = false;
+    }, 3000);
+  } catch (error) {}
```

Comment:
Can we maybe add a general toast for all the catch(error) cases so that the user actually sees that something is wrong and not just doesn't get any feedback?

Reply:
Looks good!